092-self-healing-using-cruise-control.md

```diff
@@ -0,0 +1,253 @@
+Detector classes have different mechanisms to detect their corresponding anomalies. For example, `KafkaBrokerFailureDetector` utilises Kafka Metadata API whereas `DiskFailureDetector` and `TopicAnomalyDetector` utilises Kafka Admin API. Furthermore, `MetricAnomalyDetector` and `GoalViolationDetector` uses metrics to detect the anomalies.
+The detected anomalies can be of various types:
+* Goal Violation - This happens if certain optimization goals are violated (e.g. DiskUsageDistributionGoal etc.). These goals can be configured through the `self.healing.goals` configuration under the `spec.cruiseControl.config` section. These goals are independent of the manual rebalancing goals configured in the KafkaRebalance custom resource.
+* Topic Anomaly - Can happen when one or more topics in cluster violates user-defined properties (e.g. some partitions are too large in disk).
+* Broker Failure - It happens when a non-empty broker crashes or leaves a cluster.
+* Disk Failure - This failure happens if one of the non-empty disks fails (related to a Kafka Cluster with JBOD disks).
+* Metric anomaly - This failure happens if metrics collected by Cruise Control have some anomaly in their value (e.g. a sudden rise in the log flush time metrics).
+
+The detected anomalies are inserted into a priority queue and the anomaly with the highest property is picked first to resolve.
+The anomaly detector manager calls the notifier to get an action regarding whether the anomaly should be fixed, delayed, or ignored.
+If the action is `FIX`, then the anomaly detector manager calls the classes that are required to resolve the anomaly.
+
+Anomaly detection also has various [configurations](https://github.com/linkedin/cruise-control/wiki/Configurations#anomalydetector-configurations), such as the detection interval and the anomaly notifier class, which can affect the performance of the Cruise Control server and the latency of the anomaly detection.
+
+#### Notifiers in Cruise Control
+
+Whenever anomalies are detected, Cruise Control provides the ability to notify the user regarding the detected anomalies using optional notifier classes.
+The notification sent by these classes increases the visibility of the operations that are taken by Cruise Control.
+However, the notifier class used by Cruise Control is configurable and custom notifiers can be used by setting the `anomaly.notifier.class` property.
+The notifier class returns the `action` that is going to be taken on the flagged anomaly.
+These actions can be classified into three types:
+* FIX - Start the anomaly fix
+* CHECK - Delay the anomaly fix
+* IGNORE - Ignore the anomaly fix
+
+The default `NoopNotifer` always sets the notifier action as `IGNORE` means that the detected anomaly will be silently ignored.
```

Comment:
I would also add that it doesn't send any notification to the user in any way as well.